presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestJoinQueries.java

```diff
@@ -33,11 +41,74 @@
 import static com.facebook.presto.tests.QueryTemplate.queryTemplate;
 import static com.google.common.collect.Iterables.getOnlyElement;
 import static java.lang.String.format;
+import static org.testng.Assert.assertNotEquals;
 import static org.testng.Assert.assertTrue;
 
 public abstract class AbstractTestJoinQueries
         extends AbstractTestQueryFramework
 {
+    @Test
+    public void testShuffledStatsWithInnerJoin()
+    {
+        // NOTE: only test shuffled stats with distributed query runner.
+        if (!(getQueryRunner() instanceof DistributedQueryRunner)) {
+            return;
+        }
+        DistributedQueryRunner queryRunner = (DistributedQueryRunner) getQueryRunner();
+
+        // Get the number of rows in orders table for query stats verification below.
+        String ordersCountQuery = "SELECT COUNT(*) FROM orders";
+        MaterializedRow ordersCountRow = Iterables.getOnlyElement(queryRunner.execute(ordersCountQuery));
+        int ordersFieldCount = ordersCountRow.getFieldCount();
+        assertTrue(ordersFieldCount == 1, format("Expected only one column, but got '%d'", ordersFieldCount));
+        Long numOfOrdersRows = (Long) ordersCountRow.getField(0);
+
+        // Get the number of rows in lineitem table for query stats verification below.
+        String lineitemCountQuery = "SELECT COUNT(*) FROM lineitem";
+        MaterializedRow lineitemCountRow = Iterables.getOnlyElement(queryRunner.execute(lineitemCountQuery));
+        int lineitemFieldCount = lineitemCountRow.getFieldCount();
+        assertTrue(lineitemFieldCount == 1, format("Expected only one column, but got '%d'", lineitemFieldCount));
+        Long numOfLineitemRows = (Long) lineitemCountRow.getField(0);
+
+        String joinQuery = "SELECT a.orderkey, a.orderstatus, b.linenumber FROM orders a JOIN lineitem b ON a.orderkey = b.orderkey";
+        // Set session property to enforce a hash partitioned join.
+        Session sessionWithPartitionedJoinProperty = Session.builder(getSession()).setSystemProperty(JOIN_DISTRIBUTION_TYPE, PARTITIONED.name()).setSystemProperty(JOIN_REORDERING_STRATEGY, NONE.name()).build();
+        assertQuery(sessionWithPartitionedJoinProperty, joinQuery);
+        List<QueryId> queryIdList = queryRunner.getQueryIds(joinQuery);
```

Comment:
Using SQL text as a query identifier is fragile as some other test may run the same query. Would it make sense to use executeWithQueryId instead? Or add a version of assertQuery that returns the query ID? 

```
ResultWithQueryId<MaterializedResult> resultWithQueryId = ((DistributedQueryRunner) queryRunner).executeWithQueryId(session, refreshSql);
QueryInfo queryInfo = ((DistributedQueryRunner) queryRunner).getQueryInfo(resultWithQueryId.getQueryId());
```

Reply:
Good point!